SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityS3UploaderDriver.java

```diff
@@ -256,31 +268,35 @@ private int checkUploads() {
       }
     }
 
-    for (SingularityUploader uploader : toRetry) {
-      LOG.debug("Retrying immediate uploader {}", uploader);
-      performImmediateUpload(uploader);
+    for (S3UploadMetadata uploaderMetadata : toRetry) {
+      SingularityUploader uploader = metadataToimmediateUploader.remove(uploaderMetadata);
+      if (uploader != null) {
+        LOG.debug("Retrying immediate uploader {}", uploaderMetadata);
+        performImmediateUpload(uploader);
+      } else {
+        LOG.debug("Uploader for metadata {} not found to retry, recreating", uploaderMetadata);
```

Comment:
not sure I understand this debug line. What is being re-created?

Reply:
Ah, copy-pasta error. the ,recreating should be removed, if we don't find the uploader in the map it means it somehow got added to both toRemove and toRetry, which theoretically should never happen, but better than throwing a NPE, updated the message